File: ResQNet_Project/resqnet-backend/src/main/java/com/pu_deltaforce/resqnet_backend/model/Ambulance.java
package com.pu_deltaforce.resqnet_backend.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "ambulances")
@Data
public class Ambulance {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String vehicleNumber;

    @Column(nullable = false)
    private String driverName;

    @Column(nullable = false)
    private String driverPhone;

    @Column(nullable = false)
    private String area;

    private Double currentLatitude;
    private Double currentLongitude;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AmbulanceStatus status;

    public enum AmbulanceStatus {
        AVAILABLE, ON_DUTY, MAINTENANCE
    }

    @PrePersist
    public void prePersist() {
        this.status = AmbulanceStatus.AVAILABLE;
    }
}

File: ResQNet_Project/resqnet-frontend/src/app/core/services/ambulance.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../environments/environment';

export interface Ambulance {
  id?: number;
  vehicleNumber: string;
  driverName: string;
  driverPhone: string;
  area: string;
  status: 'AVAILABLE' | 'ON_DUTY' | 'MAINTENANCE';
  currentLatitude?: number;
  currentLongitude?: number;
}

@Injectable({ providedIn: 'root' })
export class AmbulanceService {
  private apiUrl = `${environment.apiUrl}/ambulances`;

  constructor(private http: HttpClient) {}

  getAll() { return this.http.get<Ambulance[]>(this.apiUrl); }
  create(data: Ambulance) { return this.http.post<Ambulance>(this.apiUrl, data); }
  updateStatus(id: number, status: string) {
    return this.http.put(`${this.apiUrl}/${id}/status?status=${status}`, {});
  }
  getByArea(area: string) {
    return this.http.get<Ambulance[]>(`${this.apiUrl}/area/${area}`);
  }
}